Experiment_Launcher_code/ExperimentLauncher.py

```diff
@@ -11,25 +11,26 @@ def main():
 
     if settings:
         # Extract the necessary parameters from the settings
+        experiment_name = settings.get('experiment_name')
+        comport_name = settings.get('comport_name')
         num_trials = settings.get('num_trials')
-        trial_duration = settings.get('trial_duration')
+        return_time = settings.get('return_time')
         decision_time = settings.get('decision_time')
-
-        opponent_type = settings.get('opponent_type')  # This should now be an OpponentType enum value
-        opponent_strategy_1 = settings.get('opponent_strategy')
-        opponent_strategy_2 = settings.get('computer_opponent_strategy')
-        # Adjust logic based on enum
-
-        num_trials += 2
+        opponent_type = settings.get('opponent_type')
+        opponent1_strategy = settings.get('opponent1_strategy')
+        opponent2_strategy = settings.get('opponent2_strategy')
+        opponent1_probability = settings.get('opponent1_probability')
```

Comment:
is this only for the probability p cooperator case?

Reply:
Yes. I do not like the method of changing the gui based on the experiment configuration. All the fields needs to be on screen. All of them have default values that will pass the validity check. The experimenter changes what they need and ignore what is irrelevant.